vignettes/openhub_project_search.Rmd

```diff
@@ -0,0 +1,230 @@
+* html_url_or_name: Either the URL for the organization page on OpenHub (e.g. "https://openhub.net/orgs/apache") or the short, unique, handle for the organization (e.g. "apache").
+* language: The code language to filter for projects only containing the specified code language (e.g. "java").
+* token: The file named "openhub_token" containing the OpenHub API Token.
+
+## API Response Folder Storage
+
+A hardcoded function for folder creation to store the API responses from each endpoint:
+
+```{r, eval = FALSE}
+openhub_folder_create <- function(folder_path, folder_name) {
+  folder <- file.path(folder_path, folder_name)
+  if (!dir.exists(folder)) {
+    folder <- io_make_folder(folder_path, folder_name)
+  }
+  return(folder)
+}
+github_folder <- dirname(dirname(getwd()))
+rawdata_folder <- openhub_folder_create(github_folder, "rawdata")
+openhub_folder <- openhub_folder_create(rawdata_folder, "openhub")
+study_folder <- openhub_folder_create(openhub_folder, study_name)
+organization_folder_path <- openhub_folder_create(study_folder, "organization/")
+portfolio_project_folder_path <- openhub_folder_create(study_folder, "portfolio/")
+project_folder_path <- openhub_folder_create(study_folder, "project/")
+analysis_folder_path <- openhub_folder_create(study_folder, "analysis/")
+```
+
+# Collecting and Parsing Data via Ohloh API
+
+In this section, for each endpoint, we collect the data through a series of Ohloh API requests, and parse the API responses with its corresponding parser function. These parsed API responses are data tables which are displayed for each subsection. The values from one endpoint may be extracted for use to obtain a path to the next endpoint, and the merging of data tables is important for a holistic display of the data for the list of projects.
+
+## Organizations
+
+We call `openhub_api_organizations` to acquire the organization API response of our selected organization specified by `html_url_or_name`. This function downloads the API response file into the folder specified by `organization_folder_path`.
+
+```{r, eval = FALSE}
+openhub_api_organizations(token, organization_folder_path, html_url_or_name)
+```
+
+With the organization API response (only one page), we may parse this response with its corresponding parser function, `openhub_parse_organizations`, to acquire a data table with columns representing the tags for the organization listed:
+
+* name: The name of the organization.
+* html_url_projects: The URL to the organization on OpenHub's website corresponding to a list of portfolio projects for the organization.
+* portfolio_projects: The number of portfolio projects under the organization.
+
+```{r, eval = FALSE}
+openhub_organization_api_requests <- openhub_retrieve(organization_folder_path)
+openhub_organizations <- openhub_parse_organizations(openhub_organization_api_requests)
+gt(openhub_organizations)
```

Comment:
look at the gt that I did on the merged notebooks: this is missing the pipe with the head operator. Failing that this will be a massive table. @RavenMarQ @crepesAlot  @daomcgill please check your notebooks have that.

Reply:
Will be added.